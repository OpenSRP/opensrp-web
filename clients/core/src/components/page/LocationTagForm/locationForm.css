.form-background {
  background-color: #FFFFFF;
  box-shadow: 0 2px 4px 0 #D7D7E0;
  width: 1050px;
  height: 569px;
  padding: 45px;
}

.field-row {
  margin-bottom: 10px
}

.input-field {
  border: 1px solid #DADCE0;
  border-radius: 2px;
  width: 404px;
  height: 48px;
}

.btn-bg {
  background-color: #0292D3;
  width: 404px;
  height: 48px;
  color: #FFFFFF;
  border: 1px solid #DADCE0;
  margin-top: 10px;
}

.input-row {
  margin-top: 10px;
}