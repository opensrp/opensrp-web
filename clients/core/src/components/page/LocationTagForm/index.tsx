import * as React from 'react';
import { Col, Row } from 'reactstrap';
import { generateOptions } from '../../../services/opensrp';
import { OpenSRPService } from '@opensrp/server-service';
import '@opensrp/opensrp-table/dist/index.css';
import './locationForm.css';
import { Formik } from 'formik';
import { OPENSRP_API_BASE_URL, OPENSRP_LOCATION_TAG_CREATE_ENDPOINT } from '../../../configs/env';

interface LocationTagForm {
    name: string;
    description: string;
    active: boolean;
}

const defaultProps: LocationTagForm = {
    name: '',
    description: '',
    active: false,
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const FormikForm = (formik: any) => {
    return (
        <div>
            <Row>
                <Col>
                    <h3>New Location Tag</h3>
                </Col>
            </Row>
            <div className="form-background">
                <form onSubmit={formik.handleSubmit}>
                    <Row>
                        <label>Name</label>
                    </Row>
                    <Row>
                        <input
                            type="text"
                            className="input-field"
                            onChange={formik.handleChange}
                            onBlur={formik.handleBlur}
                            value={formik.values.name}
                            name="name"
                        />
                    </Row>
                    <Row className="input-row">
                        <label>Description</label>
                    </Row>
                    <Row>
                        <input
                            type="text"
                            className="input-field"
                            onChange={formik.handleChange}
                            onBlur={formik.handleBlur}
                            value={formik.values.description}
                            name="description"
                        />
                    </Row>
                    <Row>
                        <button type="submit" className="submit-btn-bg">
                            Submit
                        </button>
                    </Row>
                </form>
            </div>
        </div>
    );
};
const LocationTagForm: React.FC<LocationTagForm> = (props: LocationTagForm) => {
    const onSubmit = async (values: LocationTagForm) => {
        const payload = {
            name: values.name,
            description: values.description,
            active: values.active,
        };
        const clientService = new OpenSRPService(
            OPENSRP_API_BASE_URL,
            OPENSRP_LOCATION_TAG_CREATE_ENDPOINT,
            generateOptions,
        );
        await clientService.create(payload);
    };

    return (
        <Formik initialValues={props} onSubmit={onSubmit}>
            {FormikForm}
        </Formik>
    );
};

LocationTagForm.defaultProps = defaultProps;
export default LocationTagForm;
