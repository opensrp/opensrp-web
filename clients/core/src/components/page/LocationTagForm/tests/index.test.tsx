// test ExportModal
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import LocationTagForm from '..';

describe('components/Location-tag-form', () => {
    it('renders without crashing', () => {
        const props: LocationTagForm = {
            name: '',
            description: '',
            active: false,
        };
        shallow(<LocationTagForm {...props} />);
    });

    it('renders the Loading component', () => {
        const props: LocationTagForm = {
            name: '',
            description: '',
            active: false,
        };
        const wrapper = mount(<LocationTagForm {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    it('submit the form with values', async () => {
        const props: LocationTagForm = {
            name: 'District',
            description: 'level 1',
            active: true,
        };
        const wrapper = mount(<LocationTagForm {...props} />);
        const submitBtn = wrapper.find('.submit-btn-bg');
        expect(submitBtn.length).toEqual(1);

        submitBtn.simulate('click');
    });
});
