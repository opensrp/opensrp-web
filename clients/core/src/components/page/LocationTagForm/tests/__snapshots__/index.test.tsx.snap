// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Location-tag-form renders the Loading component 1`] = `
<LocationTagForm
  active={false}
  description=""
  name=""
>
  <Formik
    initialValues={
      Object {
        "active": false,
        "description": "",
        "name": "",
      }
    }
    onSubmit={[Function]}
  >
    <div>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="row"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col"
            >
              <h3>
                New Location Tag
              </h3>
            </div>
          </Col>
        </div>
      </Row>
      <div
        className="form-background"
      >
        <form
          onSubmit={[Function]}
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <label>
                Name
              </label>
            </div>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <input
                className="input-field"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </Row>
          <Row
            className="input-row"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="input-row row"
            >
              <label>
                Description
              </label>
            </div>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <input
                className="input-field"
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <button
                className="submit-btn-bg"
                type="submit"
              >
                Submit
              </button>
            </div>
          </Row>
        </form>
      </div>
    </div>
  </Formik>
</LocationTagForm>
`;
